src/polling.ts

```diff
@@ -90,41 +90,60 @@ export class PollingProcessor {
       });
   }
 
-  private async retrieveFlags(): Promise<void> {
-    try {
-      this.log.debug('Fetching flags started');
-      const response = await this.api.getFeatureConfig(
-        this.environment,
-        this.cluster,
-      );
-      this.log.debug('Fetching flags finished');
-      response.data.forEach((fc: FeatureConfig) =>
-        this.repository.setFlag(fc.feature, fc),
-      );
-    } catch (error) {
-      this.log.error('Error loading flags', error);
-      throw error;
+  private async retrieveFlags(retries = 3): Promise<void> {
```

Comment:
This function is a bit of a wall of text too and difficult to visually scan.

Reply:
Fixed: https://github.com/harness/ff-nodejs-server-sdk/commit/cccd3e8127c3c9e1828b73f6c774974bd111819b